src/main/java/com/palpal/dealightbe/domain/item/domain/ItemRepository.java

```diff
@@ -1,8 +1,48 @@
 package com.palpal.dealightbe.domain.item.domain;
 
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 public interface ItemRepository extends JpaRepository<Item, Long>, ItemRepositoryCustom {
 
 	boolean existsByNameAndStoreId(String name, Long storeId);
+
+	@Query("SELECT i FROM Item i JOIN i.store s " +
+		"WHERE s.storeStatus = 'OPENED'" +
+		"AND (6371 * ACOS(COS(RADIANS(:yCoordinate)) " +
+		"    	* COS(RADIANS(s.address.yCoordinate)) " +
+		"		* COS(RADIANS(s.address.xCoordinate) - RADIANS(:xCoordinate)) " +
+		"       + SIN(RADIANS(:yCoordinate)) * SIN(RADIANS(s.address.yCoordinate)))) < 3" +
+		"ORDER BY " +
+		"CASE " +
+		"  WHEN s.closeTime < CURRENT_TIME THEN " +
+		"    (24 * 60 * 60 + (HOUR(s.closeTime) * 3600 + MINUTE(s.closeTime) * 60 + SECOND(s.closeTime))) " +
+		"  ELSE " +
+		"    (HOUR(s.closeTime) * 3600 + MINUTE(s.closeTime) * 60 + SECOND(s.closeTime)) " +
+		"END ASC, i.updatedAt DESC")
```

Comment:
쿼리문 작성 하신김에 레포지토리 테스트도 하면 좋을 것 같아요! 그리고 여러줄에 거친 쿼리문은 이렇게도 작성할 수 있답니다!



```suggestion
	@Query(
	"""
	SELECT i FROM Item i JOIN i.store s 
WHERE s.storeStatus = 'OPENED'
AND (6371 * ACOS(COS(RADIANS(:yCoordinate)) 
    	* COS(RADIANS(s.address.yCoordinate)) 
		* COS(RADIANS(s.address.xCoordinate) - RADIANS(:xCoordinate)) 
       + SIN(RADIANS(:yCoordinate)) * SIN(RADIANS(s.address.yCoordinate)))) < 3
ORDER BY 
		CASE 
  WHEN s.closeTime < CURRENT_TIME THEN 
    (24 * 60 * 60 + (HOUR(s.closeTime) * 3600 + MINUTE(s.closeTime) * 60 + SECOND(s.closeTime))) 
		ELSE 
    (HOUR(s.closeTime) * 3600 + MINUTE(s.closeTime) * 60 + SECOND(s.closeTime)) 
END ASC, i.updatedAt DESC")
	"""
```